### PR TITLE
Created fan attack and 2nd enemy

### PR DESCRIPTION
**Pull request by danielkim10**
Looks like this
image

Changes:
- Created a 2nd enemy that does the fan attack and have it spawn next to the original. My main concern is that I named it EnemyAI2 instead of something more descriptive so that should be changed later. I think it's ok for it to use an identical sprite. Tags and other constants are also basically identical to the original enemy.
- Created an AttackFan script and assigned it to the 2nd enemy.
- Moved the Shoot method out of EnemyAI and into the Attack scripts, I think this will make it easier to manipulate the velocity of the missiles to create more attack patterns.

Pending improvements:
- Rotate the missile so that it corresponds with its movement trajectory.
- Along with rotating the missile, check that the collision boxes are also rotated accordingly.
- Check for lag. I felt like my simulator started to lag a little after adding the 2nd enemy but it could just be my laptop.
- The 2nd enemy is only there to show the attack, it can always be removed.

Let me know how I can make improvements to the code because implementing this felt pretty scrappy :(

### CODE REVIEW

**Review by Raymond-Xia — CHANGES_REQUESTED** [review]
image
Need to save game state !

File: Assets/Scripts/Entities/EnemyAI2.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public int ammo;
+    public Bounds missilePosition;
+    public GameObject canvas;
+    // Start is called before the first frame update
+    void Start()
+    {
+        enemy = GameObject.Find(Constants.ENEMY2_PREFAB);
+        StartCoroutine(Spawn(enemy, 5.0f));
+        enemy.SetActive(true);
+        StartCoroutine(AttackFan.Shoot(missile, 1, ammo, shootSpeed, shootPos));
```
Comment (Raymond-Xia) [inline Assets/Scripts/Entities/EnemyAI2.cs:21]: Not entirely a fan of having an entire new AI script just for a different attack pattern, is there some way we can dynamically assign different attack patterns to the same EnemyAI script?

**Review by Raymond-Xia — COMMENTED** [review]
In terms of missile direction/orientation we can just simplify it and have the hitbox be a square instead of a rectangle, the rectangle shape is mainly arbitrary anyway. Then we don't have to deal with rotation and it looks fine.

File: Assets/Scripts/Entities/EnemyAI2.cs (hunk 1 above)
Comment (danielkim10) [reply Assets/Scripts/Entities/EnemyAI2.cs]: The GameObject.SendMessage function is a pretty good substitute

**Review by Raymond-Xia — APPROVED** [review]
It is awesome now

**Pull request merged**